diff -Naur testtools-0.9.30.orig/testtools/compat.py testtools-0.9.30/testtools/compat.py
--- testtools-0.9.30.orig/testtools/compat.py	2013-05-18 02:37:20.054013319 +0000
+++ testtools-0.9.30/testtools/compat.py	2013-05-18 02:38:59.910489029 +0000
@@ -33,10 +33,7 @@
 BytesIO = try_imports(['StringIO.StringIO', 'io.BytesIO'])
 StringIO = try_imports(['StringIO.StringIO', 'io.StringIO'])
 
-try:
-    from testtools import _compat2x as _compat
-except (SyntaxError, ImportError):
-    from testtools import _compat3x as _compat
+from testtools import _compat3x as _compat
 
 reraise = _compat.reraise
 
