--- ceph-17.1.0-175-g086c8f84/src/arrow/cpp/cmake_modules/ThirdpartyToolchain.cmake.orig	2022-04-08 11:27:53.593570634 -0400
+++ ceph-17.1.0-175-g086c8f84/src/arrow/cpp/cmake_modules/ThirdpartyToolchain.cmake	2022-04-08 11:28:20.778087653 -0400
@@ -1991,7 +1991,7 @@
 
 if((NOT ARROW_SIMD_LEVEL STREQUAL "NONE") OR (NOT ARROW_RUNTIME_SIMD_LEVEL STREQUAL "NONE"
                                              ))
-  set(xsimd_SOURCE "BUNDLED")
+  set(xsimd_SOURCE "SYSTEM")
   resolve_dependency(xsimd)
   # TODO: Don't use global includes but rather target_include_directories
   include_directories(SYSTEM ${XSIMD_INCLUDE_DIR})
